scalableminds/webknossos
# improve sorting when filtering datasets

I used the dice coefficient to sort the datasets by best match. This works quite well for the suffix example which covers probably 95% of the cases where this is needed. I'm a bit concerned about performance, since the sorting resulted in a noticable lag for 1000 datasets. However, my datasets all had almost the same name. So, if the search query filters the amount of datasets considerably, the lag should be reduced, as well. In the end, the lab should just give final feedback (after deployment, since the dev server doesn't have many datasets). For the copy&paste use case it should definitely be enough.

### Mailable description of changes:
 - When using the search functionality in the datasets view, the datasets will be sorted so that the best match is shown first. If a different sorting is desired, the sorting-arrows in the columns can still be used to change the sorting criteria.

### URL of deployed dev instance (used for testing):
- https://dicesortdatasets.webknossos.xyz

### Steps to test:
- open dashboard (default sort order should be "created")
- sort by a different criteria
- type something in the search bar (blue sorting-arrow should turn gray since the datasets are sorted by best match now)
- sort again by another criteria --> should work even after refining the search query

### Issues:
- fixes #2818

------
